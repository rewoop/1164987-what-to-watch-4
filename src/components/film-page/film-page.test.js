import React from "react";
import renderer from "react-test-renderer";
import FilmPage from "./film-page.jsx";
import FilmPageOverview from "../film-page-overview/film-page-overview";

const sortedFilmsMock = () => {
  return (
    <article
      className="small-movie-card catalog__movies-card"
      onClick={() => {}}
      onMouseEnter={() => {}}
      onMouseLeave={() => {}}>
      <div className="small-movie-card__image">
        <video/>
        <img src="img/the-grand-budapest-hotel-poster.jpg" alt="The Rock film poster" width="280" height="175"/>
      </div>
      <h3 className="small-movie-card__title">
        <a onClick={() => {}} className="small-movie-card__link" href="#">The Rock</a>
      </h3>
    </article>
  );
};

const Settings = {
  FILM_TITLE: `The Rock`,
  FILM_GENRE: `Thriller`,
  FILM_SRC: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  RELEASE_DATE: 1996,
  RUN_TIME: `1h 39m`,
  BACKGROUND_POSTER: `img/bg-the-grand-budapest-hotel.jpg`,
  FILM_POSTER: `img/the-grand-budapest-hotel-poster.jpg`,
  RATING: {
    SCORE: `9.0`,
    LEVEL: `Very good`,
    COUNT: `1337 ratings`
  },
  FILM_DESCRIPTION: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  FILM_DIRECTOR: `Wes Andreson`,
  FILM_STARRING: [
    `Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Tom Cruz`
  ],
  REVIEWS: [
    {
      id: `0`,
      author: `Kate Muir`,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      date: `December 24, 2016`,
      rating: `8,9`,
    }
  ],
};

const renderActiveTab = () => {
  return <FilmPageOverview
    ratingScore={Settings.RATING.SCORE}
    ratingLevel={Settings.RATING.LEVEL}
    ratingCount={Settings.RATING.COUNT}
    filmDescription={Settings.FILM_DESCRIPTION}
    filmDirector={Settings.FILM_DIRECTOR}
    filmStarring={Settings.FILM_STARRING}
  />;
};

it(`Should FilmPage render correctly`, () => {
  const tree = renderer
    .create(<FilmPage
      filmTitle={Settings.FILM_TITLE}
      filmGenre={Settings.FILM_GENRE}
      filmReleaseDate={Settings.RELEASE_DATE}
      backgroundFilmPoster={Settings.BACKGROUND_POSTER}
      filmPoster={Settings.FILM_POSTER}
      ratingScore={Settings.RATING.SCORE}
      ratingLevel={Settings.RATING.LEVEL}
      ratingCount={Settings.RATING.COUNT}
      filmDescription={Settings.FILM_DESCRIPTION}
      filmDirector={Settings.FILM_DIRECTOR}
      filmStarring={Settings.FILM_STARRING}
      runTime={Settings.RUN_TIME}
      reviews={Settings.REVIEWS}
      sortedFilms={sortedFilmsMock()}
      activeTab={`overview`}
      renderActiveTab={renderActiveTab}
      setActiveTab={() => {}}
      filmSrc={Settings.FILM_SRC}
      onPlayButtonClickHandler={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

  expect(tree).toMatchSnapshot();
});
