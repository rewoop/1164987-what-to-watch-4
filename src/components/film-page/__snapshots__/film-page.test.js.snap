// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPage render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Rock"
          src="img/bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/login"
            onClick={[Function]}
            style={
              Object {
                "color": "inherit",
                "cursor": "pointer",
                "textDecoration": "none",
              }
            }
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Rock
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Thriller
            </span>
            <span
              className="movie-card__year"
            >
              1996
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Rock"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              9
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Very good
              </span>
              <span
                className="movie-rating__count"
              >
                1337 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Wes Andreson
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="player__video"
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            />
            <img
              alt="Fantastic Beasts"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film"
              onClick={[Function]}
            >
              Fantastic Beasts
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
